### PR TITLE
Add support for GitHub Actions cache API

### PR DESCRIPTION
**Pull request by alidevhere**
@gmlewis   Ready for review. Please mention if something needs a change.
Thanks 

Fixes: #2584.

### CODE REVIEW

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:30]: I think this one is safe to leave as `*int` (not `*int64` as we tend to reserve that for really large numbers or numeric IDs).

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:38]: We tend to add `omitempty` when are not sure if the GitHub server will populate the field...
but if we are confident that this will be a populated field, then we can get rid of the pointer and make this a `string` instead of a `*string`.

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:39]: We try to match the name of the field to avoid confusion, so this would be `ActiveCachesSizeInBytes`, and I think this one is fine to leave as an `int64`.

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:40]: I'm thinking that this one should probably be an `int`.

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:43]: "list repositories" => "list of repositories"

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:45]: Please add a space between `// GitHub` here and all other locations in this file.

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:47]: Probably `int` here.

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:51]: "github" => "GitHub"

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:55]:
```suggestion
	TotalActiveCachesSizeInBytes int64 `json:"total_active_caches_size_in_bytes"`
	TotalActiveCachesCount    int `json:"total_active_caches_count"`
```

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:64]: Instead of duplication here, let's please just change this 4 lines to `ListOption`.

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:73]: For this section, and all comments throughout, please add a single space after the `//`.

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:76]:
```suggestion
// ListCaches lists the GitHub Actions caches for a repository.
```

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:154]: Please add a blank line above line 154 for consistency.

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:159]: Please add a blank line above line 159 for consistency.

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:162]:
```suggestion
// ListCacheUsageByRepoForOrg lists repositories and their GitHub Actions cache usage for an organization. The data fetched using this API is
```

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:181]: Please insert a blank line. See above.

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:183]: Please delete this blank line. See above.

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:191]:
```suggestion
// GetTotalCacheUsageForOrg gets the total GitHub Actions cache usage for an organization. The data fetched using this API is refreshed approximately every
```

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:202]: Please delete this blank line. See above.

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:206]: Please insert a blank line. See above.

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:211]: Please insert a blank line. See above.

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:214]:
```suggestion
// GetTotalCacheUsageForEnterprise gets the total GitHub Actions cache usage for an enterprise. The data fetched using this API is refreshed approximately every 5 minutes,
```

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:235]: Please insert a blank line. See above.

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:224]: Please delete this blank line. See above.

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:231]: Please delete this blank line. See above.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @alidevhere !

**Comment by alidevhere** [comment]
@gmlewis  I have made all requested changes and re ran all tests and commands. 

Please review again and do mention if something is still missing.
Thanks  😃

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:55]:
```suggestion
	TotalActiveCachesUsageSizeInBytes int64 `json:"total_active_caches_size_in_bytes"`
```

File: github/actions_cache.go
Comment (gmlewis) [inline github/actions_cache.go:72]:
```suggestion
	Ref *string `url:"ref,omitempty"`
	Key *string `url:"key,omitempty"`
	// Can be one of: "created_at", "last_accessed_at", "size_in_bytes". Default: "last_accessed_at"
	Sort *string `url:"sort,omitempty"`
	// Can be one of: "asc", "desc" Default: desc
	Direction *string `url:"direction,omitempty"`
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @alidevhere !
Just a few more tweaks, please, then I think we will be ready for a second LGTM+Approval from any other contributor to this repo before merging.

**Review by gmlewis — APPROVED** [review]
Thank you, @alidevhere !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by valbeat — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

**Pull request merged**
